### ISSUE TITLE
Migrate to GitHub-native Dependabot

### ISSUE THREAD

**Issue by HonkingGoose** [terser/terser#841]
**Bug report or Feature request?**

I think feature request is closer to the mark than bug report.
As what you currently have is still working.

**Version (complete output of `terser -V` or specific git commit)**

Not applicable.

**Complete CLI command or `minify()` options used**

Not applicable.

**`terser` input**

Not applicable.

**`terser` output or error**

Not applicable.

**Expected result**

The project uses the GitHub-native Dependabot instead of the old Dependabot preview app.

----

### Benefits of proposed change:

- The new Dependabot is fully integrated into GitHub, no need to go to a separate Dependabot website to manage the bot.
- You no longer need to have the `dependabot-preview` app installed for this repository.
- Configuration is in a `dependabot.yml` file, so it's easy to see what the configuration of the bot is.
- Configuration is public.

### Drawbacks of proposed change:

- You lose the ability to let Dependabot auto-merge when the tests pass.
- Configuration is public.

### How to make the change:
You can migrate from the old Dependabot-preview app to the built-int Dependabot quite easily.
Dependabot will do most of the migration for you.

Follow the steps laid out in this GitHub blog post:
https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/#moving-forward-from-dependabot-com-and-dependabot-preview

If you run into problems, I'm available to help, just ping me with a mention.

**Comment by fabiosantoscode** [comment]
Hey there!

I got really annoyed at dependabot's habit of sending PRs for irrelevant patch bumps so I reconfigured it. But I guess I should remove it, because this package only has three dependencies which don't change often.

**Comment by HonkingGoose** [comment]
I you want, I can create a pull request to reconfigure Dependabot to your liking. 
I can tell it to ignore your development depencies, so that you'll only get updates for your main dependencies. For more on ignoring dependencies with Dependabot: https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/configuration-options-for-dependency-updates#ignore

Otherwise you might want to take a look at Renovatebot, I think you can configure that bot to only give you major, and/or minor level patches. That way you won't be bothered with "irrelevant patch bumps".

Let me know if you like any of this, or if you want me to shut up about it. :smile:

**Comment by fabiosantoscode** [comment]
Renovatebot seems good :) I would greatly appreciate a PR!

**Comment by HonkingGoose** [comment]
Do you want a pull request for a Renovate bot configuration or for a Dependabot configuration?

**Comment by fabiosantoscode** [comment]
For Renovate bot. 

I'll go see what's needed to get the ball rolling :)

**Comment by HonkingGoose** [comment]
You can get started yourself by:

- Removing the old `dependabot-preview` app from the list of installed apps for this repository.
- Installing the Renovate app for this repository. https://github.com/marketplace/renovate

Docs for installing the Renovate app: https://docs.renovatebot.com/install-github-app/

List of installed apps globally for all you do on GitHub can be found here, you can also turn off Dependabot for your repository there if needed: https://github.com/settings/apps/authorizations

List of installed apps for this repository: https://github.com/terser/terser/settings/installations

I think that the Renovate app creates a pull request to get your started and to explain how it works, and how to configure it.
So I would recommend first trying that and seeing how you get on.
I'm on standby to help you figure it out, as I have never used Renovate myself.

**Comment by fabiosantoscode** [comment]
Alright!

I looked into the docs a bit and configured it. Wasn't too hard at all.

Thanks for your help and for the suggestion! I was getting tired of all these PRs :)

**Comment by HonkingGoose** [comment]
Now you have a dependency bot you're happy with!

I saw the on-boarding PR from the Renovate bot, and it was very clear about what you're doing. And the configuration wasn't too difficult even.

Thanks for considering my issue, and making some changes!

**Comment by fabiosantoscode** [comment]
Yeah, the onboarding PR was very helpful. Usually when configuring something like this, where you place configuration files on your repo, a syntax error or misconfiguration takes extra iterations to fix. In this case, I could just click edit, on their commit, and view a dry-run before I merged it to enable things.

The only hitch I had with it was that, on the list of things it was going to do, it should've said what configuration variable each item belonged to.

I wanted to change a bunch of things, and it would've been nice to just read the name of the configs I had to turn off or adjust, maybe a link to the corresponding doc. (CC @whitesource have some feedback :) )
